flask_lastuser/__init__.py

```diff
@@ -218,15 +232,24 @@ def before_request(self):
                 user = self.load_user(g.lastuser_cookie['userid'])
                 if not user:
                     # As above, try to create user record
-                    user = self.lastuser.login_from_cookie(g.lastuser_cookie['userid'])
+                    user = self.lastuser.login_from_cookie(
+                        g.lastuser_cookie['userid'])
+        return user, user_from_token, token_error
+
+    def before_request(self):
+        """
+        Listener that is called at the start of each request. Responsible for
+        """
+        user, user_from_token, token_error = self.get_logged_in_user()
 
         if not user:
             g.lastuser_cookie.pop('userid', None)
             g.lastuser_cookie.pop('sessionid', None)
 
         g.user = user
+        lastuser_cache = self.get_lastuser_cache(header_only=True)
         if user:
-            g.access_scope = ['*']  # TODO: In future, restrict to access token's scope
+            g.access_scope = lastuser_cache['scope']  # TODO: In future, restrict to access token's scope
```

Comment:
If you don't have an access token but have a cookie user, `lastuser_cache` will be `None` and the scope should be `*`.

Reply:
@jace we can just check in required_login etc that if the specified scope is in `g.access_scope` mentioned here, right? no need to check anything else then.

Reply:
After making the changes you mentioned above.

Reply:
Yeah, that is what you need to be doing. Cookie login sets `g.access_scope` to `['*']` while token login sets a specific scope list.